### PR TITLE
Implemented Regex with ICU

### PR DESCRIPTION
**Pull request by ypopovych**

### CODE REVIEW

File: Regex.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -925,6 +943,7 @@
 				65C529C91C5664B0002BF559 /* Release */,
 			);
 			defaultConfigurationIsVisible = 0;
+			defaultConfigurationName = Release;
```
Comment (dileping) [inline Regex.xcodeproj/project.pbxproj:946]: Why release?

File: Regex/FoundationUtils.swift [hunk 2]
```diff
@@ -15,21 +15,20 @@
 //===----------------------------------------------------------------------===//
 
 #if !os(Linux)
-    //here we use NSRegularExpression
-    import Foundation
+import Foundation
+#endif
     
-    enum InvalidRangeError : ErrorType {
```
Comment (dileping) [inline Regex/FoundationUtils.swift:21]: What really have you changed in this file?

File: Regex.xcodeproj/project.pbxproj (hunk 1 above)
Comment (ypopovych) [inline Regex.xcodeproj/project.pbxproj:946]: Xcode changed automatically... Will fix

File: Regex/FoundationUtils.swift (hunk 2 above)
Comment (ypopovych) [inline Regex/FoundationUtils.swift:21]: Moved extension from NSRange to type alias GroupRange. And made it allowed for all platforms

**Pull request closed without merging**